server/cp/archive_search.py

```
@@ -33,231 +24,592 @@ def __len__(self):
+                        field, direction = next(iter(first.items()))
+                        fld = str(field).lower()
+                        dir_ = "asc" if str(direction).lower() == "asc" else "desc"
+                        if fld in ("versioncreated", "firstcreated"):
+                            p["sort"] = f"{fld}:{dir_}"
+
+            api_params = self._build_api_params(params=p, size=size, frm=frm)
+
+            logger.info("Final API params: %s", api_params)
+
+            items, total = self._api_search(api_params)
+            logger.info("API returned %d items of total %s", len(items), total)
+
+            return ArchiveListCursor(self._transform_items(items), total)
+        except Exception:
+            logger.exception("Archive API search failed")
+            return ArchiveListCursor([], 0)
+
+    def _headers(self) -> Dict[str, str]:
+        h = {
+            "Accept": "application/json",
+            "User-Agent": "cp-archive-search-provider/1.0",
+        }
+        if self.api_key:
+            h["x-api-key"] = self.api_key
+        return h
+
+    @staticmethod
+    def _extract_query_text_from_es(es_query: dict) -> str:
+        try:
+            q = es_query or {}
+            node = q.get("query") or {}
+            if "filtered" in node:
+                node = node["filtered"].get("query") or {}
+
+            if isinstance(node, dict):
+                if "query_string" in node and isinstance(node["query_string"], dict):
+                    s = node["query_string"].get("query")
+                    if isinstance(s, str) and s.strip():
+                        return s.strip()
+                if "simple_query_string" in node and isinstance(
+                    node["simple_query_string"], dict
+                ):
+                    s = node["simple_query_string"].get("query")
+                    if isinstance(s, str) and s.strip():
+                        return s.strip()
+        except Exception:
+            pass
```

is it supposed to silent fail here?